cpp/src/io/utilities/file_io_utilities.cpp

```diff
@@ -34,6 +34,14 @@ namespace cudf {
 namespace io {
 namespace detail {
 
+void force_init_cuda_context()
+{
+  // Workaround for https://github.com/rapidsai/cudf/issues/14140, where cuFileDriverOpen errors
+  // out if no CUDA calls have been made before it. This is a no-op if the CUDA context is already
+  // initialized
+  cudaFree(0);
```

Comment:
Why do we hard code `0`?

Reply:
> Why do we hard code `0`?

This initializes the context without actually doing anything.
https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY.html#group__CUDART__MEMORY_1ga042655cbbf3408f01061652a075e094

Reply:
As far as I can tell, `cudaFree(0)` is the idiomatic way to do this. nullptr would be a better option in general, but it looks like `cudaFree(0)` is recognized as the idiom. I haven't seen a single reference where `cudaFree(nullptr)` is used.

Reply:
This is the idiom that I am familiar with as well.